lms/djangoapps/instructor_analytics/basic.py

```diff
@@ -432,6 +432,8 @@ def dump_grading_context(course):
 
 def iterate_problem_components(course):
     """ Helper function for iterating through all problems of a course """
+
+    valid_types = getattr(settings, 'STUDENT_RESPONSES_REPORT_SUPPORTED_TYPES', [])
```

Comment:
I don't care for this level of (over-)protection. `STUDENT_RESPONSES_REPORT_SUPPORTED_TYPES` will always be defined, so we don't need to try safely grabbing it with `getattr`.

The only way `STUDENT_RESPONSES_REPORT_SUPPORTED_TYPES` would ever be undefined is if something has gone terribly wrong. In that case, we actually want it to break, to flag us of the error.